ftw/globalstatusmessage/tests/test_controlpanel.py

```diff
@@ -39,3 +57,98 @@ def test_controlpanel_removed_on_uninstall(self):
         actions = [a.getAction(self)['id']
                    for a in self.controlpanel.listActions()]
         self.assertNotIn('globalstatusmessage', actions)
+
+
+class PublishedControlPanelTestCase(FunctionalTestCase):
+
+    enabled_bool = True
+    enabled_anonymous_bool = True
+    type_choice = u'information'
+    title_textfield = u'Hello World'
+    message_textfield = u'This is the message to the world'
+    exclude_sites = None
+
+    form_data = {
+        'Active': enabled_bool,
+        'Show to anonymous users?': enabled_anonymous_bool,
+        'Type': type_choice,
+        'Title': title_textfield,
+        'Message': message_textfield,
+    }
+
+    expected_settings = {
+        'enabled_bool': enabled_bool,
+        'enabled_anonymous_bool': enabled_anonymous_bool,
+        'type_choice': type_choice,
+        'title_textfield': title_textfield,
+        'message_textfield': message_textfield,
+        'exclude_sites': exclude_sites,
+    }
+
+    payload = json.dumps(decode_for_json(expected_settings))
+
+    def setUp(self):
+        super(PublishedControlPanelTestCase, self).setUp()
+        self.grant('Manager')
+        api.portal.get_tool('portal_setup').runAllImportStepsFromProfile(
+            'profile-ftw.publisher.sender:default',
+            ignore_dependencies=True
+        )
+
+    def test_all_settings_are_tested(self):
+        """
+        This test will fail if new settings are added to "IStatusMessageConfigForm"
+        without having been added in this test case.
+        """
+        self.assertEqual(
+            set(self.expected_settings.keys()),
+            set(getFieldNames(IStatusMessageConfigForm)),
+            msg='Have you added some fields to "IStatusMessageConfigForm" without updating the test case?'
+        )
+
+    @browsing
+    def test_sender(self, browser):
+        intercepted_data = {}
+
+        class MockedReceiverView(BrowserView):
+            def __call__(self):
+                intercepted_data['jsondata'] = self.request.form.get('jsondata')
+                return createResponse(SuccessState())
+
+        config = IConfig(self.portal)
+        config.appendRealm(Realm(1, self.portal.absolute_url(), SITE_OWNER_NAME, SITE_OWNER_PASSWORD))
+        transaction.commit()
+
+        with view_registered(MockedReceiverView, 'global_statusmessage_config_receiver',
+                             required=(IPloneSiteRoot, Interface)):
+            browser.login(SITE_OWNER_NAME).open(view='@@global_statusmessage_config')
+            browser.fill(self.form_data)
+            browser.click_on('Save and publish')
+
+        self.assertDictEqual(
+            self.expected_settings,
+            encode_after_json(json.loads(intercepted_data['jsondata']))
+        )
+
+        self.assertEqual(
+            ['Changes saved and published.'],
+            info_messages()
+        )
+
+    @browsing
+    def test_receiver(self, browser):
+        browser.login().open(
```

Comment:
But the test user has `Manager` role: https://github.com/4teamwork/ftw.globalstatusmessage/pull/27/files#diff-978b2c4c1213d0eb32b198bb534d79dbR92